[docs] EAS Build - add new npm hooks (#18413)
expo/expo: docs/pages/build-reference/how-tos.md

There are three EAS Build-specific npm hooks that you can set in your package.json. See the [Android build process](android-builds.md) and [iOS build process](ios-builds.md) docs to get a better understanding about the internals of the build process.

- `eas-build-pre-install` - executed before EAS Build runs `yarn install`.
- `eas-build-post-install` - the behavior depends on the platform and project type:

>   - for managed Android project, it will run after `yarn install` and `expo prebuild` have completed.
>   - for bare Android project, it will run after `yarn install` has completed.
>   - for both bare and managed iOS projects, it will run after `yarn install` and `pod install` have completed.

Reviewer: ```suggestion
  - Android
    - managed projects - runs after `yarn install` and `expo prebuild`.
    - bare projects - runs after `yarn install`.
  - iOS - runs after `yarn install` and `pod install`.
```

- `eas-build-on-success` - this hook is triggered at the end of the build process if the build was successful.
- `eas-build-on-error` - this hook is triggered at the end of the build process if the build failed.

> - `eas-build-on-success` - this hook is triggered at the end of the build process. You can check a status of the build using `EAS_BUILD_STATUS` environment variable, it will be set to `finished` or `errored`.

Reviewer: ```suggestion
- `eas-build-on-success` - this hook is triggered at the end of the build process. You can check the build's status with the `EAS_BUILD_STATUS` environment variable. It's either `finished` or `errored`.
```
